### PR TITLE
US 02.01.01: QR code 

### LINKED ISSUE #18: US 02.01.01
As a an organizer I want to create a new event and generate a unique promotional QR code that links to the event description and event poster in the app.

Acceptance:
- [x] Navigate between two screens
- [x] QR code based on ID is generated
- [x] Connected to database
Additional:
- [x] Connect "details" button to view event details

### PR DESCRIPTION
**Pull request by SuperCreativeUniqueUsername**
US 02.01.01: As a an organizer I want to create a new event and generate a unique promotional QR code that links to the event description and event poster in the app.

CRC updated:
image

UML updated:
image

Storyboard updated:
image

Todo:
Once database connection is added this will need to be updated
Once the single event poster is done navigation would need to be updated

To navigate to the qrcode the event must be "posted" and for this an image must be uploaded to the event. I cannot figure out how to do testing for image selector, I cant find anything online for this, except for `` androidTestImplementation("androidx.test.uiautomator:uiautomator:2.2.0") `` but this too makes no sense for me. Does anyone have any idea for how to test picking an image? So therefore there is no testing done for the qrcode fragment in terms of code written. I tested this on my phone, and it seems to work and the id passed to the qrviewer matches that of the event.

### CODE REVIEW

**Review by Jameszf — APPROVED** [review]
Looks good to me!

File: app/src/main/res/layout/o_event_qrcode_view.xml [hunk 1]
```diff
@@ -0,0 +1,72 @@
+<?xml version="1.0" encoding="utf-8"?>
```
Comment (Jameszf) [inline app/src/main/res/layout/o_event_qrcode_view.xml]: Can you rename the file to use "organiser" prefix instead of "o"

Comment (SuperCreativeUniqueUsername) [reply app/src/main/res/layout/o_event_qrcode_view.xml]: I went with that since there a some other files using that convention.

Comment (SuperCreativeUniqueUsername) [reply app/src/main/res/layout/o_event_qrcode_view.xml]: I think it does make sense to change it though

File: app/src/androidTest/java/com/example/pygmyhippo/organiser/PostEventFragmentTesting.java [hunk 2]
```diff
@@ -98,12 +123,9 @@ public void testWithData() {
...
-    // I think its possible but it seems really complicated
-//    @Test
-//    public void imageUploadTest() {
-//        Uri imageUri = Uri.parse("android.resource://com.example.pygmyhippo/drawable/dog.jpg");
-//        onView(withId(R.id.o_postEvent_addImage)).perform(click());
-//
-//    }
+    @Test
+    public void testNavigationToQR() {
+
```
Comment (katzhong) [inline app/src/androidTest/java/com/example/pygmyhippo/organiser/PostEventFragmentTesting.java:112]: add comment? or delete?

File: app/src/main/java/com/example/pygmyhippo/organizer/EventQRViewerFragment.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        // set up for the details button
+        //detailsButton = view.findViewById(R.id.o_eventqr_detailsBtn);
+        //backButton.setOnClickListener(view1 -> {
+            // TODO: connect to navigating to single event
+            //Navigation.findNavController(view1).navigate(R.id.action_view_eventqr_fragment_to_organiser_postEvent_page);
+        //});
+
+        super.onCreate(savedInstanceState);
+        Bundle bundle = this.getArguments();
+        myEventString = bundle.getString("my_event_id");
```
Comment (katzhong) [inline app/src/main/java/com/example/pygmyhippo/organizer/EventQRViewerFragment.java]: more concise naming?

**Review by katzhong — APPROVED** [review]
looks good to me

**Pull request merged**